librz/core/cgraph.c

```diff
@@ -0,0 +1,755 @@
+ * \brief Get a line graph of the function blocks at \p addr.
+ */
+RZ_API RZ_OWN RzGraph /*<RzGraphNodeInfo *>*/ *rz_core_graph_line(RZ_NONNULL RzCore *core, ut64 addr) {
+	return rz_core_graph_function_bbs(core, addr, block_line);
+}
+
+/**
+ * \brief Get a normal graph of the function blocks at \p addr.
+ */
+RZ_API RZ_OWN RzGraph /*<RzGraphNodeInfo *>*/ *rz_core_graph_normal(RZ_NONNULL RzCore *core, ut64 addr) {
+	return rz_core_graph_function_bbs(core, addr, NULL);
+}
+
+/**
+ * \brief Get a graph of specific type (\p type) at \p addr.
+ */
+RZ_API RZ_OWN RzGraph /*<RzGraphNodeInfo *>*/ *rz_core_graph(RZ_NONNULL RzCore *core, RzCoreGraphType type, ut64 addr) {
+	rz_return_val_if_fail(core && core->analysis, NULL);
+	RzGraph *graph = NULL;
+	switch (type) {
+	case RZ_CORE_GRAPH_TYPE_DATAREF:
+		graph = rz_core_graph_datarefs(core, addr);
+		break;
+	case RZ_CORE_GRAPH_TYPE_FUNCALL:
+		graph = rz_core_graph_callgraph(core, addr);
+		break;
+	case RZ_CORE_GRAPH_TYPE_BLOCK_FUN:
+		graph = rz_core_graph_function(core, addr);
+		break;
+	case RZ_CORE_GRAPH_TYPE_IMPORT:
+		graph = rz_core_graph_importxrefs(core);
+		break;
+	case RZ_CORE_GRAPH_TYPE_REF:
+		graph = rz_core_graph_coderefs(core, addr);
+		break;
+	case RZ_CORE_GRAPH_TYPE_XREF:
+		graph = rz_core_graph_codexrefs(core, addr);
+		break;
+	case RZ_CORE_GRAPH_TYPE_LINE:
+		graph = rz_core_graph_line(core, addr);
+		break;
+	case RZ_CORE_GRAPH_TYPE_NORMAL:
+		graph = rz_core_graph_normal(core, addr);
+		break;
+	case RZ_CORE_GRAPH_TYPE_DIFF:
+	default:
+		rz_warn_if_reached();
+		break;
+	}
+	return graph;
+}
+
+/**
+ * \brief Convert a string to RzCoreGraphFormat.
+ */
+RZ_API RzCoreGraphFormat rz_core_graph_format_from_string(RZ_NULLABLE const char *x) {
+	const char short_opt = (char)(x && strlen(x) == 1 ? x[0] : 0);
+
+	if (strcmp(x, "ascii") == 0 || RZ_STR_ISEMPTY(x) || short_opt == ' ') {
+		return RZ_CORE_GRAPH_FORMAT_ASCII_ART;
+	} else if (strcmp(x, "cmd") == 0 || short_opt == '*') {
+		return RZ_CORE_GRAPH_FORMAT_CMD;
+	} else if (strcmp(x, "dot") == 0 || short_opt == 'd') {
+		return RZ_CORE_GRAPH_FORMAT_DOT;
+	} else if (strcmp(x, "gml") == 0 || short_opt == 'g') {
+		return RZ_CORE_GRAPH_FORMAT_GML;
+	} else if (strcmp(x, "json_disasm") == 0 || short_opt == 'J') {
+		return RZ_CORE_GRAPH_FORMAT_JSON_DISASM;
+	} else if (strcmp(x, "json") == 0 || short_opt == 'j') {
+		return RZ_CORE_GRAPH_FORMAT_JSON;
+	} else if (strcmp(x, "sdb") == 0 || short_opt == 'k') {
+		return RZ_CORE_GRAPH_FORMAT_SDB;
+	} else if (strcmp(x, "tiny") == 0 || short_opt == 't') {
+		return RZ_CORE_GRAPH_FORMAT_TINY;
+	} else if (strcmp(x, "interactive") == 0 || short_opt == 'v') {
+		return RZ_CORE_GRAPH_FORMAT_VISUAL;
+	}
+
+	RZ_LOG_ERROR("invalid graph format: %s\n", x);
+	return RZ_CORE_GRAPH_FORMAT_UNK;
+}
+
+/**
+ * \brief Convert a string to RzCoreGraphType.
+ */
+RZ_API RzCoreGraphType rz_core_graph_type_from_string(RZ_NULLABLE const char *x) {
+	rz_return_val_if_fail(x, RZ_CORE_GRAPH_TYPE_UNK);
```

Comment:
You say "nullable" but have an assert, remove either, but don't combine together